ambariclient/models.py

```diff
@@ -887,12 +887,23 @@ def execute_action(self, action, context, parameters=None, hosts=None):
         return self.request
 
     def decommission(self, service, hosts):
-        """Decommission slave components on a cluster.
+        self.commission(service, hosts, "decommission")
+
+    def recommission(self, service, hosts):
+        self.commission(service, hosts, "recommission")
+        """Nodemanager recommission does not start nodemanager process on the host. so its required to do it separately"""
+        if service is "YARN":
```

Comment:
@jimbobhickville, didnt noticed that my reply was still pending (until I submitted it just now) , btw, these operations are tested on my end and it works as I described. I also had chance to get it confirmed from HortonWorks guys.